File: notes_api_service/src/domains/note/note_repository.nim
from ./infrastructure/entities/note_entity import NoteEntity
from ./note_gateway import
  getNotesFromDatabase,
  getNoteFromDatabase,
  createNoteOnDatabase,
  updateNoteOnDatabase,
  deleteNoteOnDatabase

proc getNotes*(userId: int): seq[NoteEntity] =
  return getNotesFromDatabase(userId)

proc getNote*(id: int, userId: int): NoteEntity =
  return getNoteFromDatabase(id, userId)

proc createNote*(note: NoteEntity): void =
  createNoteOnDatabase(note)

proc updateNote*(note: NoteEntity): void =
  updateNoteOnDatabase(note)

proc deleteNote*(id: int, userId: int): void =
  deleteNoteOnDatabase(id, userId)

File: in_memory_cache_service/src/domains/note/note_repository.ts
import NoteEntity from "./infrastructure/entities/note_entities";
import NoteActiveRecord from "./note_active_record";

class NoteRepository {
  private noteActiveRecord: NoteActiveRecord;

  constructor(noteActiveRecord: NoteActiveRecord) {
    this.noteActiveRecord = noteActiveRecord;
  }

  public getNotes(userId: number): Promise<Array<NoteEntity>> {
    return this.noteActiveRecord.getAllNotesFromInMemoryDatabase(userId);
  }

  public getNote(noteId: number): Promise<NoteEntity> {
    return this.noteActiveRecord.getNoteFromInMemoryDatabase(noteId);
  }

  public async createNote(note: NoteEntity): Promise<void> {
    await this.noteActiveRecord.createNoteOnInMemoryDatabase(note);
  }

  public async updateNote(note: NoteEntity): Promise<void> {
    await this.noteActiveRecord.updateNoteOnInMemoryDatabase(note);
  }

  public async deleteNote(noteId: number, userId: number): Promise<void> {
    await this.noteActiveRecord.deleteNoteOnInMemoryDatabase(noteId, userId);
  }
}

export default NoteRepository;
